### DISCUSSION TITLE
Allow blank answer in a rich-text-editor question

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#6342] [Q&A] [answered]
I'd like to allow my students to submit nothing in a `pl-rich-text-editor` question:

<img width="869" alt="Screen Shot 2022-09-18 at 11 24 19 PM" src="https://user-images.githubusercontent.com/2507459/190960090-d79d0816-1e9d-420f-b67e-126779e61a0a.png">

However, students see an 'invalid, not gradeable` when they leave it blank. Of course this doesn't have any consequences since the question is worth 0 points, but still I'd like to know how to remove the 'invalid` format error.

I suppose a better, more general way, would be to submit a PR with a parameter `allow-blank` like in other pl-elements, but in the meantime, I know there's a way to do this using `server.py` but I can't quite get it to work.

I know that a blank submission (`""`) works out to be (after the encoding business): `ZnNkZ2RmZwoK`
Replacing a blank submission with N/A works out to be: `Ti9BCgo=`

I tried this in `server.py`:

```
def parse(data):
    
    if data["format_errors"] == {}:
        pass
    else:
        data['submitted_answers']['_files'] = [
            {
                'name': 'anything_else.md',
                'contents': "Ti9BCgo="
            }
        ]
```

Rather than ` if data["format_errors"] == {}:`, I also tried: `if data["submitted_answers"]['_files'][0]['contents'] == 'ZnNkZ2RmZwoK':` but I got a KeyError:

```
KeyError: '_files'
Traceback (most recent call last):
```

Any ideas? At this point, I know I'm making this too complicated.

**Comment by jonatanschroeder** [comment, marked as answer]
You are changing the content, but not clearing the error. Try:
```python
def parse(data):
    if '_files' in data['format_errors']:
        del data['format_errors']['_files']
```

And it's probably a good idea to have an issue for adding `allow-blank` to `pl-rich-text-editor` (and maybe `pl-file-editor` as well).

**Reply by firasm** [reply]
Thanks! I thought the error would be removed if I removed the offending text, but I think I would have needed to do the replacement at an earlier stage (and I'm not even sure if that's possible).

Here was my final `parse(data)` function:

```
def parse(data):
    
    if '_files' in data['format_errors']:
        del data['format_errors']['_files']
        data['submitted_answers']['_files'] = [
            {
                'name': 'anything_else.md',
                'contents': "Ti9BCgo="
            }
        ]
```

which gives me this if a student leaves the last question blank:

![Screen Shot 2022-09-20 at 12 30 06 PM](https://user-images.githubusercontent.com/2507459/191347717-002690e1-ab4c-497f-95ad-7eeda8bafab2.png)
